src/header/common/strict_transport_security.rs

```diff
@@ -39,43 +40,108 @@ use header::{Header, Raw, parsing};
 /// let mut headers = Headers::new();
 ///
 /// headers.set(
-///    StrictTransportSecurity::including_subdomains(31536000u64)
+///    StrictTransportSecurity::new()
+///         .set_max_age(StrictTransportSecurity::preload_min_age())
+///         .set_include_subdomains(true)
+///         .set_preload(true)
+///         .unwrap()
 /// );
 /// # }
 /// ```
 #[derive(Clone, PartialEq, Debug)]
 pub struct StrictTransportSecurity {
     /// Signals the UA that the HSTS Policy applies to this HSTS Host as well as
     /// any subdomains of the host's domain name.
-    pub include_subdomains: bool,
+    include_subdomains: bool,
 
     /// Specifies the number of seconds, after the reception of the STS header
     /// field, during which the UA regards the host (from whom the message was
     /// received) as a Known HSTS Host.
-    pub max_age: u64
+    max_age: u64,
+
+    /// Indicates that the server wants to be included into the HSTS preload list,
+    /// this value can be set to true only if the site:
+    /// 
+    /// * Serves a valid certificate.
+    /// * Redirects all of the traffic on port 80 to port 443.
+    /// * Serves all of its subdomains via HTTPS.
+    /// * Sets in the HSTS headers the parameter includeSubDomains and has max-age
+    ///   bigger than 31536000 seconds (one year).
+    /// * Any redirect will still be served with a HSTS header
+    preload: bool
 }
 
+/// This value indicates the minimum value of the max_age
+/// that can be used togheter with preload
+const PRELOAD_MIN_AGE: u64 = 31536000;
+
 impl StrictTransportSecurity {
-    /// Create an STS header that includes subdomains
-    pub fn including_subdomains(max_age: u64) -> StrictTransportSecurity {
+    /// Create an STS header with a expiration date
+    /// of 5 minutes with no subdomains or preload
+    pub fn new() -> StrictTransportSecurity {
         StrictTransportSecurity {
-            max_age: max_age,
-            include_subdomains: true
+            max_age: 300,
+            include_subdomains: false,
+            preload: false
         }
     }
 
-    /// Create an STS header that excludes subdomains
-    pub fn excluding_subdomains(max_age: u64) -> StrictTransportSecurity {
+    /// Returns the inserted max age
+    pub fn max_age(&self) -> u64 { self.max_age }
+
+    /// Sets the max age
+    pub fn set_max_age(&self, max_age: u64) -> StrictTransportSecurity {
```

Comment:
It's more usual for setters to just mutate the type itself, than to return a new type. We should probably stick with the usual :D